### ISSUE TITLE
Large-curve road corners clip through terrain

### ISSUE THREAD

**Issue by LeftofZen** [OpenLoco/OpenLoco#2747]
**Version information:**
 - Platform: Win11
 - Version: 24.10
 - Build/commit: master/release

**Describe the bug**
Large-curve road corners clip through terrain

**To Reproduce**
Build terrain/road as in screenshot

**Expected behaviour**
No clipping

**Screenshots**
![Image](https://github.com/user-attachments/assets/2fb6f783-51bd-479d-9a28-48f3bbc1cc87)

**Additional context**

**Comment by AaronVanGeffen** [comment]
Could this have a cause similar to #2655?

**Comment by LeftofZen** [comment]
> Could this have a cause similar to [#2655](https://github.com/OpenLoco/OpenLoco/issues/2655)?

Hmm definitely would be a similar issue, but not identical since I'm on master and have that specific fix applied

**Comment by LeftofZen** [comment]
Also visible on the rotation
![Image](https://github.com/user-attachments/assets/40951e0e-b4b5-4fe6-bc80-6acca647da94)

**Comment by ZehMatt** [comment]
Is this really a rendering issue? It seems more like it shouldn't be possible to build roads like that unless the terrain is flattened or are there specific sprites for this setup?

**Comment by LeftofZen** [comment]
> Is this really a rendering issue? It seems more like it shouldn't be possible to build roads like that unless the terrain is flattened or are there specific sprites for this setup?

well RCT2 allows you to do it (and renders correctly). don't see why it shouldn't be allowed
![Image](https://github.com/user-attachments/assets/8e6e0d60-4769-4975-aa87-bbb48754a75c)

**Comment by LeeSpork** [comment]
Original Locomotion (Steam)
![Image](https://github.com/user-attachments/assets/c7d25595-6f1c-4c89-be2f-fbd965f468a3)

**Comment by LeftofZen** [comment]
> Original Locomotion (Steam) ![Image](https://github.com/user-attachments/assets/c7d25595-6f1c-4c89-be2f-fbd965f468a3)

Thanks for checking

**Comment by LeeSpork** [comment]
Also affects train and tram tracks (the tooltip refers to this curve size as "Small Radius")

**Comment by ZehMatt** [comment]
> > Is this really a rendering issue? It seems more like it shouldn't be possible to build roads like that unless the terrain is flattened or are there specific sprites for this setup?
> 
> well RCT2 allows you to do it (and renders correctly). don't see why it shouldn't be allowed ![Image](https://github.com/user-attachments/assets/8e6e0d60-4769-4975-aa87-bbb48754a75c)

That looks quite off in my opinion, I also wouldn't exactly call over-drawing "correct" when in reality the track would be inside the slope.

**Comment by LeftofZen** [comment]
> That looks quite off in my opinion, I also wouldn't exactly call over-drawing "correct" when in reality the track would be inside the slope.

I agree with you that it looks off, but overdrawing is still better than the current situation where it's just glitchy.

"Correct" is hard to achieve in isometric perspective, as you know. The only correct thing here is to not allow placement of the curve, but how does this work in maps with the pieces already placed? People will not be happy and probably just cheat it back in. Re-rendering track+land tile to make it visually correct is also infeasible.

Fixing the rendering to match ORCT2 is the only viable solution that I see. What else do you propose?

**Comment by duncanspumpkin** [comment]
Track/surface rendering is quite different compared to rct2. It could be very difficult to achieve. Rct2 doesn't handle cliff edges like this and this is where the rendering issue is coming from.

**Comment by ZehMatt** [comment]
> > That looks quite off in my opinion, I also wouldn't exactly call over-drawing "correct" when in reality the track would be inside the slope.
> 
> I agree with you that it looks off, but overdrawing is still better than the current situation where it's just glitchy.
> 
> "Correct" is hard to achieve in isometric perspective, as you know. The only correct thing here is to not allow placement of the curve, but how does this work in maps with the pieces already placed? People will not be happy and probably just cheat it back in. Re-rendering track+land tile to make it visually correct is also infeasible.
> 
> Fixing the rendering to match ORCT2 is the only viable solution that I see. What else do you propose?

Given to what @duncanspumpkin said I would propose we disallow building tracks on slopes, I know this won't help existing saves but they can also just flatten the surface and the problem will disappear.

**Comment by LeeSpork** [comment]
> well RCT2 allows you to do it (and renders correctly). don't see why it shouldn't be allowed

OpenRCT2 renders it "incorrectly" in one rotation, interestingly
![Image](https://github.com/user-attachments/assets/25f4d5dd-5c00-4168-aeac-e31bf0cbfe9d)

**Comment by LeftofZen** [comment]
Hmm fair enough, on some of the rotations it is very 'off', to the point it looks like a bug/glitch itself
